staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -200,10 +194,31 @@ func (g *Cloud) ensureInternalLoadBalancer(clusterName, clusterID string, svc *v
 	if options.AllowGlobalAccess {
 		newFwdRule.AllowGlobalAccess = options.AllowGlobalAccess
 	}
-	if err := g.ensureInternalForwardingRule(existingFwdRule, newFwdRule); err != nil {
+
+	fwdRuleDeleted := false
+	if existingFwdRule != nil && !forwardingRulesEqual(existingFwdRule, newFwdRule) {
```

Comment:
I have some concern regarding this line. If there is any bug in the logic generating newFwdRule, this is likely to cause a delete/create loop for the forwarding rules.

The bug can potentially due to 1. a new version of controller added a new field 2. GCE defaulting behavior change etc

Feels a bit risky to compare the whole forwarding rule. Does it make sense to do spot checks? For instance, check the protocol and port and etc....

Reply:
Not sure how was this handled before?

Reply:
This is the same code from before - https://github.com/kubernetes/kubernetes/blob/a34f1e483104bd51c3e9a6aec3dbbcf6301789da/staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go#L143

Reply:
forwardingRulesEqual compares select fields - https://github.com/kubernetes/kubernetes/blob/a34f1e483104bd51c3e9a6aec3dbbcf6301789da/staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go#L750